headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/OptionCategoryResourceImpl.java

```diff
@@ -106,7 +103,44 @@ public Response patchOptionCategory(Long id, OptionCategory optionCategory)
 	public OptionCategory postOptionCategory(OptionCategory optionCategory)
 		throws Exception {
 
-		return _upsertOptionCategory(optionCategory);
+		DTOConverter optionCategoryDTOConverter =
+			_dtoConverterRegistry.getDTOConverter(
+				CPOptionCategory.class.getName());
+
+		CPOptionCategory cpOptionCategory = null;
+
+		if (optionCategory.getId() != null) {
+			cpOptionCategory = _cpOptionCategoryService.fetchCPOptionCategory(
+				optionCategory.getId());
+		}
+
+		if (cpOptionCategory == null) {
+			cpOptionCategory = _addOptionCategory(optionCategory);
+
+			return (OptionCategory)optionCategoryDTOConverter.toDTO(
+				new DefaultDTOConverterContext(
+					contextAcceptLanguage.getPreferredLocale(),
+					cpOptionCategory.getCPOptionCategoryId()));
+		}
+
+		cpOptionCategory = _updateOptionCategory(
+			optionCategory.getId(), optionCategory);
```

Comment:
can we put this in an else block, and remove the `return` statement in the `if` block (since it's identical to the one at the end of the method)? I think that'd read a little easier and hopefully be a little more maintainable.